static/skin/index.js

```diff
@@ -267,35 +279,38 @@
         });
     }
 
+    function noResultsContent(kiwixHomeBody) {
+        const divTag = document.createElement('div');
+        divTag.setAttribute('class', 'noResults');
+        divTag.innerHTML = `No result. Would you like to <a href="/?lang=">reset filter</a>?`;
+        kiwixHomeBody.append(divTag);
+        kiwixHomeBody.setAttribute('style', 'display: flex; justify-content: center; align-items: center');
+        divTag.getElementsByTagName('a')[0].onclick = (event) => {
+            event.preventDefault();
+            window.history.pushState({}, null, `${window.location.href.split('?')[0]}?lang=`);
+            setCookie(filterCookieName, 'lang=');
+            resetAndFilter();
+            removeTagElement(false);
+            document.querySelectorAll('.filter').forEach(filter => {
+                filter.value = params.get(filter.name) || '';
+                if (filter.value) {
+                    filter.style = 'background-color: #858585; color: #fff';
+                } else {
+                    filter.style = 'background-color: #ffffff; color: black';
+                }
+            })
```

Comment:
Why is this at all needed?

Reply:
I don't know it fully myself but after a series of tests, this is helping in back-forward actions.
Without this, if "reset filters" link is pressed, it correctly clears all filters. If then history back button is pressed, it correctly goes back (although I have to press it twice) but now I can't press the history forward button (gets disabled) to get back to the page with filters removed.

Reply:
I will have to deal with history tracking in #716, so I am now going to dive into https://html.spec.whatwg.org/multipage/history.html. If I find out the reason for the issues you observed with simpler code and ways to solve them easily, I will let you know.